File: ios/StrainEase/Services/FirebaseBootstrap.swift
import FirebaseCore
import Foundation
import GoogleSignIn

enum FirebaseBootstrap {
    /// iOS app registered in the same Firebase project as the web client.
    /// The iOS SDK rejects web `GOOGLE_APP_ID`s (`:web:`) — that abort is
    /// what crashed launch. Prefer `GoogleService-Info.plist` in the bundle.
    static let apiKey = "AIzaSyAVeoQkXYi3eMRaINsvDBNvbKX2XrTatBM"
    static let projectID = "strainfinder-84a9b"
    static let gcmSenderID = "89734321536"
    static let googleAppID = "1:89734321536:ios:4d986c4b721a9b28e31d65"
    static let googleClientID = "89734321536-s3njeabohn98bd8s0rqh05diur2mk9h2.apps.googleusercontent.com"

    static private(set) var isConfigured = false

    static func configure() {
        guard FirebaseApp.app() == nil else {
            isConfigured = true
            configureGoogle()
            return
        }

        let hasPlist = Bundle.main.path(forResource: "GoogleService-Info", ofType: "plist") != nil
        if hasPlist {
            FirebaseApp.configure()
        } else if googleAppID.contains(":ios:") {
            let options = FirebaseOptions(googleAppID: googleAppID, gcmSenderID: gcmSenderID)
            options.apiKey = apiKey
            options.projectID = projectID
            options.bundleID = Bundle.main.bundleIdentifier ?? "ai.strainease.app"
            options.clientID = googleClientID
            FirebaseApp.configure(options: options)
        } else {
            isConfigured = false
            return
        }

        isConfigured = FirebaseApp.app() != nil
        configureGoogle()
    }

    private static func configureGoogle() {
        let clientID = FirebaseApp.app()?.options.clientID ?? googleClientID
        GIDSignIn.sharedInstance.configuration = GIDConfiguration(clientID: clientID)
    }
}

File: android/app/src/main/java/ai/strainease/app/services/FirebaseBootstrap.kt
package ai.strainease.app.services

import android.content.Context
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions

/**
 * Firebase init that mirrors the iOS `FirebaseBootstrap.swift` so the
 * same project, the same API key, and the same Google client ID are
 * shared between Android, iOS, and the web client. The Android SDK
 * rejects web `GOOGLE_APP_ID` values (those end in `:web:`); we use
 * the Android variant registered for the iOS bundle id
 * `ai.strainease.app` since they share the Firebase project.
 *
 * PR-A4 will replace [configure] with a real `google-services.json`
 * lookup; until then, we fall back to the inline options so the app
 * boots without crashing on launch.
 *
 * Usage is a one-liner in [ai.strainease.app.StrainEaseApplication]:
 *   `FirebaseBootstrap.configure(this)`
 */
object FirebaseBootstrap {
    // Values match `android/app/google-services.json` for the
    // `ai.strainease.app` Android client registered against the
    // shared `strainfinder-84a9b` Firebase project.
    const val API_KEY = "AIzaSyAPKxwr0P4xLYijch3cFa2g0ldhqelc0aA"
    const val PROJECT_ID = "strainfinder-84a9b"
    const val GCM_SENDER_ID = "89734321536"
    const val GOOGLE_APP_ID = "1:89734321536:android:279b629e98727a43e31d65"
    const val GOOGLE_CLIENT_ID = "89734321536-l42t2nlfigvngag1so9stlok42cbp8o2.apps.googleusercontent.com"
    const val STORAGE_BUCKET = "strainfinder-84a9b.firebasestorage.app"

    @Volatile
    var isConfigured: Boolean = false
        private set

    fun configure(context: Context) {
        if (FirebaseApp.getApps(context).isNotEmpty()) {
            isConfigured = true
            return
        }
        val options = FirebaseOptions.Builder()
            .setApiKey(API_KEY)
            .setApplicationId(GOOGLE_APP_ID)
            .setProjectId(PROJECT_ID)
            .setGcmSenderId(GCM_SENDER_ID)
            .setStorageBucket(STORAGE_BUCKET)
            .build()
        FirebaseApp.initializeApp(context, options)
        isConfigured = FirebaseApp.getApps(context).isNotEmpty()
    }
}
